src/os/query/queryutils.js

```diff
@@ -1,56 +1,113 @@
+  [180, 90],
+  [180, 0],
+  [180, -89.99999999999999],
+  [-179.9999999999999, -89.99999999999999]
+]];
+
+
+/**
+ * Polygon representing the whole world.
+ * @type {ol.geom.Polygon}
+ */
+const WORLD_GEOM = new Polygon(WORLD_COORDS);
+
+
+/**
+ * @type {number|undefined}
+ * @private
+ */
+let worldArea_ = undefined;
 
 
 /**
  * Checks if an existing geometry is of type "world query"
  *
  * @param {ol.geom.Geometry|undefined} geometry The geometry to verify.
- * @return {boolean} true the query matches os.query.WORLD_GEOM
+ * @return {boolean} true the query matches WORLD_GEOM
  */
-os.query.utils.isWorldQuery = function(geometry) {
-  var world = os.query.utils.WORLD_GEOM;
-  if (world && geometry && geometry instanceof ol.geom.Polygon) {
+exports.isWorldQuery = function(geometry) {
+  if (worldArea_ == null) {
+    exports.initWorldArea();
+  }
+
+  if (worldArea_ && geometry && geometry.getType() === ol.geom.GeometryType.POLYGON) {
     // transform the world extent to the current projection to compute the area
-    var worldExtent = ol.proj.transformExtent(os.query.utils.WORLD_EXTENT, os.proj.EPSG4326, os.map.PROJECTION);
-    var worldArea = ol.extent.getArea(worldExtent);
-    if (goog.math.nearlyEquals(geometry.getArea(), worldArea) || geometry.getArea() == 0) {
-      geometry.setCoordinates(world.getCoordinates());
-      return true;
-    }
+    var geomArea = /** @type {ol.geom.Polygon} */ (geometry).getArea();
+    return googmath.nearlyEquals(geomArea / worldArea_, 1, 1E-4) || geomArea == 0;
   }
 
   return false;
 };
 
 
 /**
- * The world extent in EPSG:4326. This is the max precision that a polygon can handle.
- * @type {ol.Extent}
- * @const
+ * calculates world area
+ *
+ * @param {boolean=} opt_reset
  */
-os.query.utils.WORLD_EXTENT = [-179.9999999999999, -89.99999999999999, 180, 90];
+exports.initWorldArea = function(opt_reset) {
+  if (opt_reset) {
+    worldArea_ = undefined;
+  } else {
+    var worldExtent = olproj.transformExtent(WORLD_EXTENT, osproj.EPSG4326, osmap.PROJECTION);
+    worldArea_ = olextent.getArea(worldExtent);
+  }
+};
 
 
 /**
- * Polygon representing the whole world.
- * @type {ol.geom.Polygon}
+ * Feature representing the whole world.
+ * @type {olFeature}
  */
-os.query.utils.WORLD_GEOM = ol.geom.Polygon.fromExtent(os.query.utils.WORLD_EXTENT);
+exports.WORLD_AREA = new olFeature({
+  'geometry': WORLD_GEOM,
+  'title': 'Whole World'
+});
 
 
 /**
- * Feature representing the whole world.
- * @type {ol.Feature}
+ * Feature representing the area we want to zoom to when zooming to the whole world.
+ * @type {olFeature}
  */
-os.query.utils.WORLD_AREA = new ol.Feature({
-  'geometry': os.query.utils.WORLD_GEOM,
-  'title': 'Whole World'
-});
+exports.WORLD_ZOOM_FEATURE = new olFeature(new Polygon([[
+  [179, 90],
+  [181, 90],
+  [181, -90],
+  [179, -90],
+  [179, 90]
+]]));
+
+
+exports.WORLD_EXTENT = WORLD_EXTENT;
+exports.WORLD_COORDS = WORLD_COORDS;
+exports.WORLD_GEOM = WORLD_GEOM;
```

Comment:
Not worth holding this up over, but a cleaner way to define the exports is to just declare everything as `const/let` in the file and then define the `exports = { ... }` object down here instead of a mix.